src/components/data-table/ha-data-table.ts

```diff
@@ -830,9 +842,13 @@ export class HaDataTable extends LitElement {
     }
 
     this.sortColumn = this.sortDirection === null ? undefined : columnId;
+    if (!this.sortColumn) {
+      this.sortDirection = this._defaultSortDirection;
+      this.sortColumn = this._defaultSortColumn;
+    }
 
     fireEvent(this, "sorting-changed", {
-      column: columnId,
+      column: this.sortColumn,
```

Comment:
This changes what the event represents. Not sure if it's ok

Reply:
wrong comment, replaced it, will think again